features/all-positions/PositionActionsDialog.tsx

```diff
@@ -0,0 +1,583 @@
+
+import {
+  Box,
+  Grid,
+  Container,
+  Typography,
+  Dialog,
+  Button,
+  TextField,
+  Select,
+  InputLabel,
+  FormControl,
+  MenuItem,
+} from "@material-ui/core";
+
+import ToggleButton from "@material-ui/lab/ToggleButton";
+import ToggleButtonGroup from "@material-ui/lab/ToggleButtonGroup";
+
+import EmpState from "../../containers/EmpState";
+import DvmState from "../../containers/DvmState";
+import Token from "../../containers/Token";
+import EmpSponsors from "../../containers/EmpSponsors";
+import EmpContract from "../../containers/EmpContract";
+import Collateral from "../../containers/Collateral";
+import PriceFeed from "../../containers/PriceFeed";
+import Etherscan from "../../containers/Etherscan";
+
+const Label = styled.span`
+  color: #999999;
+`;
+
+const Status = styled(Typography)`
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+`;
+
+const PositionDialog = styled.div`
+  width: 100%;
+  border-color: white;
+  padding-left: 15px;
+  padding-right: 15px;
+  padding-bottom: 20px;
+`;
+
+const Important = styled(Typography)`
+  color: red;
+  background: black;
+  display: inline-block;
+`;
+
+const Link = styled.a`
+  color: white;
+  font-size: 14px;
+`;
+
+interface DialogProps {
+  handleClose(): any;
+  isDialogShowing: boolean;
+  selectedSponsor: string | null;
+}
+
+const PositionActionsDialog = (props: DialogProps) => {
+  const { empState } = EmpState.useContainer();
+  const { dvmState } = DvmState.useContainer();
+  const { finalFee } = dvmState;
+  const { getEtherscanUrl } = Etherscan.useContainer();
+  const { contract: emp } = EmpContract.useContainer();
+  const { latestPrice } = PriceFeed.useContainer();
+  const {
+    symbol: tokenSymbol,
+    balance: tokenBalance,
+    allowance: tokenAllowance,
+    setMaxAllowance: setMaxTokenAllowance,
+  } = Token.useContainer();
+  const {
+    symbol: collSymbol,
+    balance: collBalance,
+    allowance: collAllowance,
+    setMaxAllowance: setMaxCollateralAllowance,
+  } = Collateral.useContainer();
+  const { activeSponsors } = EmpSponsors.useContainer();
+  const [tabIndex, setTabIndex] = useState<string | null>("deposit");
+  const [collateralToDeposit, setCollateralToDeposit] = useState<string>("");
+  const [minCollPerToken, setMinCollPerToken] = useState<string>("");
+  const [maxCollPerToken, setMaxCollPerToken] = useState<string>("");
+  const [maxTokensToLiquidate, setMaxTokensToLiquidate] = useState<string>("");
+  const [deadline, setDeadline] = useState<string>("");
+
+  const [hash, setHash] = useState<string | null>(null);
+  const [success, setSuccess] = useState<boolean | null>(null);
+  const [error, setError] = useState<Error | null>(null);
+
+  const {
+    priceIdentifier: priceId,
+    collateralRequirement: collReq,
+    currentTime,
+  } = empState;
+
+  const setDialogTab = (
+    event: MouseEvent<HTMLElement>,
+    newAlignment: string | null
+  ) => {
+    setTabIndex(newAlignment);
```

Comment:
OOC: Why do we need to alloow `tabIndex` to possibly be type `null`?

Reply:
good point. have restricted to be string only.